implementations.yaml

```diff
@@ -176,7 +192,7 @@ signs:
       - name: dilithium3
         implementation: clean
         avx2_implementation: avx2
-      - name: dilithium4
+      - name: dilithium5
         implementation: clean
         avx2_implementation: avx2
   falcon:
```

Comment:
Falcon also got a bugfix and should get a minor version bump (I can't make a note on the line below).

Reply:
Falcon version number also updated.